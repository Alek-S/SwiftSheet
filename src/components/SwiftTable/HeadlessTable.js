import React, { useState } from 'react';
import PropTypes from 'prop-types';
import AutoSizer from 'react-virtualized/dist/commonjs/AutoSizer';
import Grid from 'react-virtualized/dist/commonjs/Grid';
import styled from 'styled-components';
import DefaultStyle from './StyledTable';

const list = [
	{ name: 'Brian Vaughn', description: 'Software engineer' },
	{ name: 'Alek Shnayder', description: 'FED Dev' },
	// And so on...
];

const HeadlessTable = ({ data }) => {
	console.log('SwiftTable', data);

	const _cellRenderer = ({ columnIndex, key, rowIndex, style }) => {
		return (
			<StyledCell key={key} style={style} columnIndex={columnIndex}>
				{data[rowIndex][columnIndex]}
			</StyledCell>
		);
	};

	return (
		<Styledtable>
			<AutoSizer disableHeight>
				{({ width }) => (
					<Grid
						cellRenderer={_cellRenderer}
						width={width}
						height={300}
						headerHeight={40}
						rowHeight={50}
						rowCount={data.length}
						columnCount={data[0].length}
						columnWidth={175}
					/>
				)}
			</AutoSizer>
		</Styledtable>
	);
};

HeadlessTable.propTypes = {
	data: PropTypes.arrayOf(PropTypes.object).isRequired,
};

const StyledCell = styled.div`
	display: flex;
	align-items: center;
	padding-left: 0.5rem;
	border-bottom: 1px solid ${props => props.theme.color.border};
	border-left: ${props =>
		props.columnIndex !== 0 ? `1px solid ${props.theme.color.border}` : 'none'};
`;

const Styledtable = styled(DefaultStyle)`
	margin: 3rem;
	border: 1px solid ${props => props.theme.color.border};
	border-radius: 7px;
	overflow: hidden;
	background-color: white;

	.ReactVirtualized__Table__rowColumn {
		height: 25px;
	}

	.ReactVirtualized__Table__headerRow {
		background-color: ${props => props.theme.color.blue};
		/* border-radius: 7px; */
	}
`;

export default HeadlessTable;
