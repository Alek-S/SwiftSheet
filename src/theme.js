/**
 * default themes for use with styled-components
 * passed in with theme provider
 */
const theme = {
	boxShadow: `0px 5px 40px -5px rgba(0,0,0,0.3)`,
	boxShadowDark: `0px 5px 40px -2px rgba(0,0,0,0.5)`,
	color: {
		background: '#fafafa',
		backgroundDark: '#EAEAEA',
		backgroundDarkest: '#BCBCBC',
		border: '#d9d9d9',
		text: '#404040',
		lightText: '#848484',
		red: '#ED5050',
		lightRed: '#FF8D8D',
		green: '#D8E13F',
		blue: '#46C8F5',
		lightBlue: '#97E4FF',
		darkBlue: '#2E8AA8',
	},
	gradient: {
		light: `linear-gradient(to bottom, #ffffff 0%, #f6f6f6 47%, #ededed 100%)`,
		red: 'linear-gradient(to bottom, #f07878 0%, #ED5050 100%)',
		lightRed: 'linear-gradient(to bottom, #f39191 0%, #ef6262 100%)',
		blue: 'linear-gradient(to bottom, #D8E13F 0%, #BFC644 100%)',
		greenBlue: ' linear-gradient(160deg, #57DACB 0%, #55A2ED 100%)',
	},
	font: {
		header: `'Poiret One', cursive`,
		main: `'Quicksand', sans-serif`,
	},
};

export default theme;
