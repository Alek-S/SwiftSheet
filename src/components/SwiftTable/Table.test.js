import React from 'react';
import 'jest-styled-components';

import HeadlessTable from './Table';
import { ThemeProvider } from 'styled-components';
import theme from '../../theme';

const match = {
	params: 'testId',
};
const data = [
	{
		'First Name': 'John',
		'Last Name': 'Doe',
		Address: '120 jefferson st.',
		City: 'Riverside',
		State: ' NJ',
		Zip: '8075',
	},
	{
		'First Name': 'Jack',
		'Last Name': 'McGinnis',
		Address: '220 hobo Av.',
		City: 'Phila',
		State: ' PA',
		Zip: '9119',
	},
	{
		'First Name': 'John "Da Man"',
		'Last Name': 'Repici',
		Address: '120 Jefferson St.',
		City: 'Riverside',
		State: ' NJ',
		Zip: '8075',
	},
	{
		'First Name': 'Stephen',
		'Last Name': 'Tyler',
		Address: '7452 Terrace "At the Plaza" road',
		City: 'SomeTown',
		State: 'SD',
		Zip: '91234',
	},
	{
		'First Name': '',
		'Last Name': 'Blankman',
		Address: '',
		City: 'SomeTown',
		State: ' SD',
		Zip: '298',
	},
	{
		'First Name': 'Joan "the bone", Anne',
		'Last Name': 'Jet',
		Address: '9th, at Terrace plc',
		City: 'Desert City',
		State: 'CO',
		Zip: '123',
	},
];

describe('<HeadlessTable />', () => {
	it('matches the snapshot', () => {
		const tree = mount(
			<ThemeProvider theme={theme}>
				<HeadlessTable data={data} />
			</ThemeProvider>
		);
		expect(toJson(tree)).toMatchSnapshot();
	});
});
