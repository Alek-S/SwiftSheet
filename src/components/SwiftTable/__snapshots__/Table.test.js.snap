// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeadlessTable /> matches the snapshot 1`] = `
.c0 {
  height: 90%;
  min-height: 200px;
  border: 1px solid #d9d9d9;
  font-family: 'Quicksand',sans-serif;
  font-size: 14px;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  border-radius: 2px;
  overflow: hidden;
  background-color: white;
  width: 80%;
  max-width: 1350px;
  margin: auto;
  margin-top: 20px;
}

.c0 .ag-header-cell-text {
  font-family: 'Quicksand',sans-serif;
  font-size: 14px;
}

.c0 .ag-row-even {
  background-color: #fafafa;
}

.c0 .ag-row-hover {
  background-color: #97E4FF;
}

.c0 .ag-header {
  background-color: #404040;
  color: white;
}

.c0 .ag-header .ag-column-hover {
  background-color: #fafafa;
  color: #404040;
  border-bottom: 5px solid #46C8F5;
}

.c0 .ag-icon-desc,
.c0 .ag-icon-asc {
  background-color: #46C8F5;
  border-radius: 10px;
  position: relative;
  top: 5px;
  padding: 2px;
}

.c0 .ag-icon-filter {
  background-color: #46C8F5;
  border-radius: 4px;
  position: relative;
  top: 5px;
  padding: 5px;
}

<ThemeProvider
  theme={
    Object {
      "boxShadow": "0px 5px 40px -5px rgba(0,0,0,0.3)",
      "boxShadowDark": "0px 5px 40px -2px rgba(0,0,0,0.5)",
      "boxShadowLight": "0px 5px 40px -5px rgba(0,0,0,0.15)",
      "color": Object {
        "background": "#fafafa",
        "backgroundDark": "#EAEAEA",
        "backgroundDarkest": "#BCBCBC",
        "blue": "#46C8F5",
        "border": "#d9d9d9",
        "darkBlue": "#2E8AA8",
        "green": "#D8E13F",
        "input": "#F1F3F4",
        "lightBlue": "#97E4FF",
        "lightRed": "#FF8D8D",
        "lightText": "#848484",
        "red": "#ED5050",
        "text": "#404040",
        "vividGreen": "#E3FC52",
      },
      "font": Object {
        "header": "'Poiret One', cursive",
        "main": "'Quicksand', sans-serif",
      },
      "gradient": Object {
        "blue": "linear-gradient(to bottom, #D8E13F 0%, #BFC644 100%)",
        "greenBlue": " linear-gradient(160deg, #57DACB 0%, #55A2ED 100%)",
        "light": "linear-gradient(to bottom, #ffffff 0%, #f6f6f6 47%, #ededed 100%)",
        "lightRed": "linear-gradient(to bottom, #f39191 0%, #ef6262 100%)",
        "red": "linear-gradient(to bottom, #f07878 0%, #ED5050 100%)",
      },
    }
  }
>
  <HeadlessTable
    data={
      Array [
        Object {
          "Address": "120 jefferson st.",
          "City": "Riverside",
          "First Name": "John",
          "Last Name": "Doe",
          "State": " NJ",
          "Zip": "8075",
        },
        Object {
          "Address": "220 hobo Av.",
          "City": "Phila",
          "First Name": "Jack",
          "Last Name": "McGinnis",
          "State": " PA",
          "Zip": "9119",
        },
        Object {
          "Address": "120 Jefferson St.",
          "City": "Riverside",
          "First Name": "John \\"Da Man\\"",
          "Last Name": "Repici",
          "State": " NJ",
          "Zip": "8075",
        },
        Object {
          "Address": "7452 Terrace \\"At the Plaza\\" road",
          "City": "SomeTown",
          "First Name": "Stephen",
          "Last Name": "Tyler",
          "State": "SD",
          "Zip": "91234",
        },
        Object {
          "Address": "",
          "City": "SomeTown",
          "First Name": "",
          "Last Name": "Blankman",
          "State": " SD",
          "Zip": "298",
        },
        Object {
          "Address": "9th, at Terrace plc",
          "City": "Desert City",
          "First Name": "Joan \\"the bone\\", Anne",
          "Last Name": "Jet",
          "State": "CO",
          "Zip": "123",
        },
      ]
    }
  >
    <Table__Styledtable
      className="ag-theme-material"
    >
      <div
        className="ag-theme-material c0"
      >
        <AgGridReact
          columnDefs={
            Array [
              Object {
                "field": "First Name",
                "headerName": "First Name",
              },
              Object {
                "field": "Last Name",
                "headerName": "Last Name",
              },
              Object {
                "field": "Address",
                "headerName": "Address",
              },
              Object {
                "field": "City",
                "headerName": "City",
              },
              Object {
                "field": "State",
                "headerName": "State",
              },
              Object {
                "field": "Zip",
                "headerName": "Zip",
              },
            ]
          }
          defaultColDef={
            Object {
              "columnDrag": true,
              "filter": true,
              "resizable": true,
              "sortable": true,
              "width": 225,
            }
          }
          rowData={
            Array [
              Object {
                "Address": "120 jefferson st.",
                "City": "Riverside",
                "First Name": "John",
                "Last Name": "Doe",
                "State": " NJ",
                "Zip": "8075",
              },
              Object {
                "Address": "220 hobo Av.",
                "City": "Phila",
                "First Name": "Jack",
                "Last Name": "McGinnis",
                "State": " PA",
                "Zip": "9119",
              },
              Object {
                "Address": "120 Jefferson St.",
                "City": "Riverside",
                "First Name": "John \\"Da Man\\"",
                "Last Name": "Repici",
                "State": " NJ",
                "Zip": "8075",
              },
              Object {
                "Address": "7452 Terrace \\"At the Plaza\\" road",
                "City": "SomeTown",
                "First Name": "Stephen",
                "Last Name": "Tyler",
                "State": "SD",
                "Zip": "91234",
              },
              Object {
                "Address": "",
                "City": "SomeTown",
                "First Name": "",
                "Last Name": "Blankman",
                "State": " SD",
                "Zip": "298",
              },
              Object {
                "Address": "9th, at Terrace plc",
                "City": "Desert City",
                "First Name": "Joan \\"the bone\\", Anne",
                "Last Name": "Jet",
                "State": "CO",
                "Zip": "123",
              },
            ]
          }
        >
          <div
            style={
              Object {
                "height": "100%",
              }
            }
          />
        </AgGridReact>
      </div>
    </Table__Styledtable>
  </HeadlessTable>
</ThemeProvider>
`;
